kernel/src/process/clone.rs

```diff
@@ -52,45 +53,91 @@ bitflags! {
     }
 }
 
-#[derive(Debug, Clone, Copy)]
+/// An internal structure to homogenize the arguments for `clone` and
+/// `clone3`.
+///
+/// From the clone(2) man page:
+///
+/// ```
+/// The following table shows the equivalence between the arguments
+/// of clone() and the fields in the clone_args argument supplied to
+/// clone3():
+///     clone()         clone3()        Notes
+///                     cl_args field
+///     flags & ~0xff   flags           For most flags; details
+///                                     below
+///     parent_tid      pidfd           See CLONE_PIDFD
+///     child_tid       child_tid       See CLONE_CHILD_SETTID
+///     parent_tid      parent_tid      See CLONE_PARENT_SETTID
+///     flags & 0xff    exit_signal
+///     stack           stack
+///     ---             stack_size
+///     tls             tls             See CLONE_SETTLS
+///     ---             set_tid         See below for details
+///     ---             set_tid_size
+///     ---             cgroup          See CLONE_INTO_CGROUP
+/// ```
+#[derive(Debug, Clone, Copy, Default)]
 pub struct CloneArgs {
-    new_sp: u64,
-    stack_size: usize,
-    parent_tidptr: Vaddr,
-    child_tidptr: Vaddr,
-    tls: u64,
-    clone_flags: CloneFlags,
+    pub flags: CloneFlags,
+    pub _pidfd: Option<u64>,
+    pub child_tid: Vaddr,
+    pub parent_tid: Option<Vaddr>,
+    pub exit_signal: Option<SigNum>,
+    pub stack: u64,
+    pub stack_size: Option<NonZeroU64>,
+    pub tls: u64,
+    pub _set_tid: Option<u64>,
+    pub _set_tid_size: Option<u64>,
+    pub _cgroup: Option<u64>,
 }
```

Comment:
The problem with this is that it makes the construction via `From<Clone3Args>` somewhat awkward, as we would need to do something like:

```rust
let mut args = CloneArgs::default();
args.set_flags(flags);
args.set_pidfd(pidfd);
// ...and so son for each field
```

For the record, the fields for this struct were already public, I simply expanded it.

Reply:
Alright, I'm fine with the current change if you believe that making these fields public is a better approach, especially since this struct will only be constructed once and is unlikely to be misused.

> For the record, the fields for this struct were already public, I simply expanded it.

Sorry, I didn't get it. The fields of `CloneArgs` were all private before this PR.

Reply:
> > For the record, the fields for this struct were already public, I simply expanded it.
> 
> Sorry, I didn't get it. The fields of `CloneArgs` were all private before this PR.

Ah you're right, I was thinking of something else. Anyhow, I'll mark this as resolved.